aws-lc-rs/src/cipher.rs

```diff
@@ -3,158 +3,1270 @@
+            let aes_256_key_bytes =
+                from_hex("000102030405060708090a0b0c0d0e0f000102030405060708090a0b0c0d0e0f")
+                    .unwrap();
+            let cipher_key = UnboundCipherKey::new(&AES_256, aes_256_key_bytes.as_slice()).unwrap();
+            assert_eq!("UnboundCipherKey { algorithm: Algorithm { id: Aes256, key_len: 32, block_len: 16 } }", format!("{cipher_key:?}"));
+        }
+
+        {
+            let key_bytes = &[0u8; 16];
+            let key = PaddedBlockEncryptingKey::cbc_pkcs7(
+                UnboundCipherKey::new(&AES_128, key_bytes).unwrap(),
+            )
+            .unwrap();
+            assert_eq!("PaddedBlockEncryptingKey { key: UnboundCipherKey { algorithm: Algorithm { id: Aes128, key_len: 16, block_len: 16 } }, mode: CBC, padding: PKCS7, context: Iv128 }", format!("{key:?}"));
+            let mut data = vec![0u8; 16];
+            let context = key.encrypt(&mut data).unwrap();
+            assert_eq!("Iv128", format!("{context:?}"));
+            let key = PaddedBlockDecryptingKey::cbc_pkcs7(
+                UnboundCipherKey::new(&AES_128, key_bytes).unwrap(),
+                context,
+            )
+            .unwrap();
+            assert_eq!("PaddedBlockDecryptingKey { key: UnboundCipherKey { algorithm: Algorithm { id: Aes128, key_len: 16, block_len: 16 } }, mode: CBC, padding: PKCS7, context: Iv128 }", format!("{key:?}"));
+        }
+
+        {
+            let key_bytes = &[0u8; 16];
+            let key =
+                EncryptingKey::ctr(UnboundCipherKey::new(&AES_128, key_bytes).unwrap()).unwrap();
+            assert_eq!("EncryptingKey { key: UnboundCipherKey { algorithm: Algorithm { id: Aes128, key_len: 16, block_len: 16 } }, mode: CTR, context: Iv128 }", format!("{key:?}"));
+            let mut data = vec![0u8; 16];
+            let context = key.encrypt(&mut data).unwrap();
+            assert_eq!("Iv128", format!("{context:?}"));
+            let key =
+                DecryptingKey::ctr(UnboundCipherKey::new(&AES_128, key_bytes).unwrap(), context)
+                    .unwrap();
+            assert_eq!("DecryptingKey { key: UnboundCipherKey { algorithm: Algorithm { id: Aes128, key_len: 16, block_len: 16 } }, mode: CTR, context: Iv128 }", format!("{key:?}"));
+        }
+    }
+
+    fn helper_test_cipher_n_bytes(
+        key: &[u8],
+        alg: &'static Algorithm,
+        mode: OperatingMode,
+        n: usize,
+    ) {
+        let mut input: Vec<u8> = Vec::with_capacity(n);
+        for i in 0..n {
+            let byte: u8 = i.try_into().unwrap();
+            input.push(byte);
+        }
+
+        let cipher_key = UnboundCipherKey::new(alg, key).unwrap();
+        let encrypting_key = EncryptingKey::new(cipher_key, mode, None).unwrap();
+
+        let mut in_out = input.clone();
+        let decrypt_iv = encrypting_key.encrypt(&mut in_out).unwrap();
+
+        if n > 5 {
+            // There's no more than a 1 in 2^48 chance that this will fail randomly
```

Comment:
This is a test where we originally didn't have the `n > 5` condition, so with 1 byte there was about a (1/256) probability that the ciphertext would be the same as the plaintext and the next test condition would fail.  For n bytes, the probability of a match is around 1/(2^(n*8)).